convokit/storage/storageManager.py

```diff
@@ -0,0 +1,212 @@
+                for collection_name in value.list_collection_names():
+                    warn(f'\tDropping collection {collection_name} (collection in storage.{attr}))')
+                    value.drop_collection(collection_name)
+
+    @staticmethod
+    def default_storage_mode():
+        tmp_mode = os.environ.get('CONVOKIT_STORAGE_MODE')
+        if tmp_mode in ['mem', 'db']:
+            return tmp_mode
+        else:
+            config = read_or_create_config(
+                os.path.expanduser('~/.convokit/config.yml'))
+            return config['default_storage_mode']
+
+    @staticmethod
+    def purge_db():
+        """
+        Remove all data in the default database. 
+        """
+        warn('purging the DB')
+        client = MongoClient()
+        db = client['convokit']
+        for collection_name in db.list_collection_names():
+            warn(
+                f'\tDropping collection {collection_name} (collection in {db.name}))'
+            )
+            db.drop_collection(collection_name)
+        client.close()
+
+    def setup_collections(self, utterance, conversation, speaker,
+                          convokitmeta):
+        """
+        Setup instance variables self._utterances, self._conversations, self._speakers & self._metas.
+        Should be called as storageManager.setup_collections(Utterance, Conversation, Speaker, ConvoKitMeta)
+        after initializing the storageManager:StorageManager.
+        Utterance, Conversation, Speaker, ConvoKitMeta cannot be imported in the convokit.storage module due 
+        to circular imports, so a call to setup_collections is required to configure a StorageManager with 
+        the correct types. 
+        """
+        self._utterances = self.CollectionMapping('utterances',
+                                                  item_type=utterance)
+        self._conversations = self.CollectionMapping('conversations',
+                                                     item_type=conversation)
+        self._speakers = self.CollectionMapping('speakers', item_type=speaker)
+        self._metas = self.CollectionMapping('metas', item_type=convokitmeta)
+
+    def __eq__(self, other):
+        if not isinstance(other, StorageManager):
+            return False
+        else:
+            return self.storage_type == other.storage_type \
+                and self.full_name == other.full_name
+
+    def __repr__(self):
+        return f'<StorageManager: {self.storage_type}-{self.full_name}>'
+
+
+def safe_corpus_id():
+    return str(randrange(2**15, 2**20))
+
+
+def make_full_name(corpus_id, version):
+    return f'{corpus_id}_v{version}'
+
+
+def read_or_create_config(config_fullpath):
+    if not os.path.isfile(config_fullpath):
+        convo_dir = os.path.expanduser('~/.convokit')
+        if not os.path.isdir(convo_dir):
+            os.makedirs(convo_dir)
+        with open(config_fullpath, 'w') as f:
+            text = ("# Default Storage Parameters\n"
```

Comment:
For easier maintenance this text should be defined as a constant at the top of the file